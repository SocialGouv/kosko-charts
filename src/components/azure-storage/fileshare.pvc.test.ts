import { ResourceRequirements } from "kubernetes-models/v1";

import { fileSharePersistentVolumeClaim } from "./fileshare.pvc";

jest.mock("@socialgouv/kosko-charts/environments/gitlab", () => ({
  // eslint-disable-next-line @typescript-eslint/naming-convention
  __esModule: true,
  default: () => ({
    manifest: {
      annotations: {
        "app.gitlab.com/app": "socialgouv-sample",
        "app.gitlab.com/env": "my-test",
      },
      labels: {
        application: "sample",
        owner: "sample",
        team: "sample",
      },
      namespace: { name: "sample-42-my-test" },
    },
    projectName: "sample",
    shortSha: "abcdefg",
  }),
}));

beforeEach(() => {
  jest.resetModules();
});

test("should create a persistent volume claim and a persistent volume", () => {
  const pvc = fileSharePersistentVolumeClaim({
    metadata: { name: "foobar" },
    resources: new ResourceRequirements({ requests: { storage: "5Gi" } }),
  });
  expect(pvc).toMatchSnapshot();
  expect(() => {
    pvc.validate();
  }).not.toThrow();
});
