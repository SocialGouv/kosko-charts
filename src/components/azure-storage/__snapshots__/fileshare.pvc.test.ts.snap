// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create a persistent volume claim and a persistent volume 1`] = `
Object {
  "apiVersion": "v1",
  "kind": "PersistentVolumeClaim",
  "metadata": Object {
    "annotations": Object {
      "app.gitlab.com/app": "socialgouv-sample",
      "app.gitlab.com/env": "my-test",
    },
    "labels": Object {
      "application": "sample",
      "owner": "sample",
      "team": "sample",
    },
    "name": "foobar",
    "namespace": "sample-42-my-test",
  },
  "spec": Object {
    "accessModes": Array [
      "ReadWriteMany",
    ],
    "resources": Object {
      "requests": Object {
        "storage": "5Gi",
      },
    },
    "storageClassName": "azurefile",
  },
}
`;
